src/ngircd/irc-channel.c

```diff
@@ -264,8 +313,27 @@ join_send_topic(CLIENT *Client, CLIENT *target, CHANNEL *chan,
 	if (Client_Type(Client) != CLIENT_USER)
 		return true;
 	/* acknowledge join */
+	#ifndef IMPLEMENT_EXTENDED_JOIN
 	if (!IRC_WriteStrClientPrefix(Client, target, "JOIN :%s", channame))
 		return false;
+	#endif
+	#ifdef IMPLEMENT_EXTENDED_JOIN
+	CONN_ID conn;
+	conn = Client_Conn(Client);
+	if(Client_Cap(Client) & CLIENT_CAP_EXTENDED_JOIN) {
+		char * account_name;
+		if(Client_AccountName(Client)) {
```

Comment:
Please don't add braces where they are not needed (`{ one_directive; }`).
Just use:
```
if(…)
    directive;
```